kernels/python/Dockerfile

```diff
@@ -0,0 +1,102 @@
+# syntax = docker/dockerfile:1.2.1
+ARG PYTHON_VERSION
+FROM jupyter/datascience-notebook:python-${PYTHON_VERSION}
+
+USER root
+
+# datascience-notebook:python-3.9.13 includes psutil 5.9.2 with cooked C lib, but
+# later pip installs end up installing 5.9.4, but for some
+# reason 'import psutil' will end up getting the python 5.9.4 but the
+# C lib from 5.9.2, and, unlike Smeagol, it hateses the precious.
+RUN pip uninstall -y psutil
+
+# Set up log file for magics
+RUN touch /var/log/noteable_magics.log && \
+  chown 4004:4004 /var/log/noteable_magics.log
+
+# When image is run, run the code with the environment
+# activated:
+SHELL ["/bin/bash", "-c"]
+
+WORKDIR /tmp
+
+# hadolint ignore=DL3008,DL3015
+RUN apt-get update && \
+  apt-get install -y jq procps git unixodbc-dev g++ \
+  && rm -rf /var/lib/apt/lists/*
+
+ENV NB_USER="noteable" \
+  NB_UID=4004 \
+  NB_GID=4004
+
+# Create the default unprivileged user
+RUN groupadd --gid 4004 noteable && \
+  useradd --uid 4004 --shell /bin/false --create-home --no-log-init --gid noteable noteable && \
+  chown --recursive noteable:noteable /home/noteable
+
+RUN mkdir /etc/ipython && chown noteable:noteable /etc/ipython
```

Comment:
I'm not sure why we decided to use the `/etc/` directory as the base of operations for the notebook files. `/etc` is really meant more for system configuration files. Currently, sysdig also flags this as an issue given the the privileges generally required to modify these files and it being suspicious for doing so.  Since this is going to be our base golden image, we should get this conversation started with the backend folks. I recommend we move it to `/srv` instead.